### PR TITLE
refactor: Use union type for SkipReason

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Replace enum with string union type
- Change `'non-hex depTypes'` to `'non-hex-dep-types'`

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — DISMISSED** [review]

File: lib/manager/helmfile/extract.ts [hunk 1]
```diff
@@ -43,15 +42,15 @@ export function extractPackageFile(
...
+          skipReason: 'invalid-name',
         };
       }
 
       // If starts with ./ is for sure a local path
       if (dep.chart.startsWith('./')) {
         return {
           depName,
-          skipReason: SkipReason.LocalChart,
+          skipReason: 'local-chart',
```
Comment (viceice) [inline lib/manager/helmfile/extract.ts:53]:
```suggestion
          skipReason: 'local-dependency',
```

Should we unify this with others?

File: lib/manager/jenkins/extract.ts [hunk 2]
```diff
@@ -21,30 +20,30 @@ function getDependency(plugin: JenkinsPlugin): PackageDependency {
...
     plugin.source?.version === 'experimental' ||
     plugin.groupId
   ) {
-    dep.skipReason = SkipReason.UnsupportedVersion;
+    dep.skipReason = 'unsupported-version';
   }
 
   if (plugin.source?.url) {
-    dep.skipReason = SkipReason.InternalPackage;
+    dep.skipReason = 'internal-package';
```
Comment (viceice) [inline lib/manager/jenkins/extract.ts:42]:
```suggestion
    dep.skipReason = 'local-dependency';
```

Same here?

File: lib/manager/npm/extract/index.ts [hunk 3]
```diff
@@ -259,23 +258,23 @@ export async function extractPackageFile(
       }
     }
     if (dep.currentValue.startsWith('file:')) {
-      dep.skipReason = SkipReason.File;
+      dep.skipReason = 'file';
```
Comment (viceice) [inline lib/manager/npm/extract/index.ts:261]:
```suggestion
      dep.skipReason = 'local-dependency';
```

File: lib/manager/npm/extract/monorepo.ts [hunk 4]
```diff
@@ -42,7 +41,7 @@ export async function detectMonorepos(
       if (!updateInternalDeps) {
         p.deps?.forEach((dep) => {
           if (internalPackageNames.includes(dep.depName)) {
-            dep.skipReason = SkipReason.InternalPackage;
+            dep.skipReason = 'internal-package';
```
Comment (viceice) [inline lib/manager/npm/extract/monorepo.ts:44]:
```suggestion
            dep.skipReason = 'local-dependency';
```

File: lib/manager/npm/extract/monorepo.ts [hunk 5]
```diff
@@ -61,7 +60,7 @@ export async function detectMonorepos(
         if (!updateInternalDeps) {
           subPackage.deps?.forEach((dep) => {
             if (internalPackageNames.includes(dep.depName)) {
-              dep.skipReason = SkipReason.InternalPackage;
+              dep.skipReason = 'internal-package';
```
Comment (viceice) [inline lib/manager/npm/extract/monorepo.ts:63]:
```suggestion
              dep.skipReason = 'local-dependency';
```

File: lib/manager/terraform/modules.ts [hunk 6]
```diff
@@ -69,7 +68,7 @@ export function analyseTerraformModule(dep: PackageDependency): void {
   } else if (dep.managerData.source) {
     const moduleParts = dep.managerData.source.split('//')[0].split('/');
     if (moduleParts[0] === '..') {
-      dep.skipReason = SkipReason.Local;
+      dep.skipReason = 'local';
```
Comment (viceice) [inline lib/manager/terraform/modules.ts:71]:
```suggestion
      dep.skipReason = 'local-dependency';
```

File: lib/manager/terraform/resources.ts [hunk 7]
```diff
@@ -89,15 +88,15 @@ export function analyseTerraformResource(
...
       }
       break;
 
     case TerraformResourceTypes.helm_release:
       if (!dep.managerData.chart) {
-        dep.skipReason = SkipReason.InvalidName;
+        dep.skipReason = 'invalid-name';
       } else if (checkIfStringIsPath(dep.managerData.chart)) {
-        dep.skipReason = SkipReason.LocalChart;
+        dep.skipReason = 'local-chart';
```
Comment (viceice) [inline lib/manager/terraform/resources.ts:99]:
```suggestion
        dep.skipReason = 'local-dependency';
```

File: lib/manager/terragrunt/modules.ts [hunk 8]
```diff
@@ -56,7 +55,7 @@ export function analyseTerragruntModule(dep: PackageDependency): void {
   } else if (dep.managerData.source) {
     const moduleParts = dep.managerData.source.split('//')[0].split('/');
     if (moduleParts[0] === '..') {
-      dep.skipReason = SkipReason.Local;
+      dep.skipReason = 'local';
```
Comment (viceice) [inline lib/manager/terragrunt/modules.ts:58]:
```suggestion
      dep.skipReason = 'local-dependency';
```

File: lib/types/skip-reason.ts [hunk 9]
```diff
@@ -1,46 +1,45 @@
...
-  UnsupportedDatasource = 'unsupported-datasource',
-  UnsupportedRemote = 'unsupported-remote',
-  UnsupportedUrl = 'unsupported-url',
-  UnsupportedVersion = 'unsupported-version',
-  Unsupported = 'unsupported',
-  UnversionedReference = 'unversioned-reference',
-  VersionPlaceholder = 'version-placeholder',
-  IsPinned = 'is-pinned',
-}
+export type SkipReason =
```
Comment (viceice) [inline lib/types/skip-reason.ts:1]: i think we should unify some values like:
* `internal-package` -> `local-dependency`
* `local-chart` -> `local-dependency`
* `local` -> `local-dependency`

Can be done in separate PR of cause.

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Comment by zharinov** [comment]
@viceice Thanks for suggestions, will unify these as the separate task

**Review by viceice — APPROVED** [review]

**Pull request merged**